src/main/java/org/junit/internal/requests/MemoizingRequest.java

```diff
@@ -8,7 +8,7 @@
 
 abstract class MemoizingRequest extends Request {
     private final Lock runnerLock = new ReentrantLock();
-    private volatile Runner runner;
```

Comment:
The fields in JUnit that are volatile are intentionally volatile. https://www.baeldung.com/java-volatile describes a good explanation as to why we do this.